athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/S3BlockSpillReader.java

```diff
@@ -75,7 +78,7 @@ public Block read(S3SpillLocation spillLocation, EncryptionKey key, Schema schem
             return block;
         }
         catch (IOException ex) {
-            throw new RuntimeException(ex);
+            throw new AthenaConnectorException(ex, ex.getMessage(), ErrorDetails.builder().errorCode(FederationSourceErrorCode.INTERNAL_SERVICE_EXCEPTION.toString()).build());
```

Comment:
This probably shouldn't be an internal service error. I'm wondering if we should introduce a RuntimeException to FederationSourceErrorCode.

Reply:
took a look at the available exceptions from Glue. Lets keep this as a service error and we'll fix later

Reply:
sure.